src/edge/simulator.ts

```diff
@@ -91,85 +52,94 @@ export class Simulator {
-            const items: vscode.MessageItem[] = [ install ];
-            if (!force) {
-                items.push({ title: Constants.skipForNow });
-            }
-
-            let input: vscode.MessageItem | undefined;
-            if (force) {
-                input = await vscode.window.showWarningMessage(message, { modal: true }, ...items);
-            } else {
-                input = await vscode.window.showWarningMessage(message, ...items);
-            }
-
-            if (input === install) {
-                try {
-                    await Executor.executeCMD(outputChannel, "pip", {shell: true}, "install --upgrade iotedgehubdev");
-                } catch (error) {
-                    if (outputChannel) {
-                        outputChannel.appendLine(`${Constants.failedInstallSimulator} ${error.message}`);
-                    }
-                    ret =  InstallReturn.Failed;
-                }
-            } else {
-                ret = InstallReturn.Canceled;
-            }
-        } else {
-            ret = InstallReturn.NoPip;
-        }
-        return ret;
+    private static async getLastestStandaloneSimulatorInfo() {
+        const infoString = await request.get(Simulator.latestReleaseInfoUrl, {
+            headers: {
+                "User-Agent": "vscode-azure-iot-edge",
+            },
+        });
+        const infoJson = JSON.parse(infoString);
+        return infoJson;
     }
 
-    private static async simulatorInstalled(): Promise<boolean> {
-        return await Simulator.checkCmdExist("iotedgehubdev");
+    private static adjustTerminalCommand(command: string): string {
+        return (os.platform() === "linux" || os.platform() === "darwin") ? `sudo ${command}` : command;
     }
 
-    private static async checkPipInstalled(): Promise<boolean> {
-        return await Simulator.checkCmdExist("pip");
+    private isInstalling: boolean = false;
+
+    constructor(private context: vscode.ExtensionContext) {
     }
 
-    private static async checkCmdExist(cmd: string): Promise<boolean> {
+    public async validateSimulatorUpdated(outputChannel: vscode.OutputChannel = null): Promise<void> {
+        let message: string;
+        let type: string = "";
+        const telemetryName = "simulatorUpdated";
         try {
-            await Executor.executeCMD(null, cmd, { shell: true }, "--version");
-            return true;
+            const simulatorType = await this.getSimulatorType();
+            if (simulatorType === SimulatorType.NotInstalled) {
+                message =  Constants.needSimulatorInstalledMsg;
+                type = "install";
+            } else {
+                const version: string | null = await this.getCurrentSimulatorVersion();
+                if (version && semver.valid(version)) {
+                    const latestVersion: string | undefined = await this.getLatestSimulatorVersion();
+                    if (latestVersion && !semver.eq(latestVersion, version)) {
```

Comment:
why not gt?

Reply:
If user configured the version, need to check version?

Reply:
Update this logic